model/src/cfd.rs

```diff
@@ -296,6 +298,35 @@ impl Order {
             self.opening_fee,
         )
     }
+
+    const OUTDATED_AFTER_MINS: i64 = 10;
+
+    /// Defines when we consider the order to be sane within the system boundaries
+    ///
+    /// This is a safety net for the taker to prevent the taker from taking an outdated order.
+    pub fn is_safe_to_take(&self, now: OffsetDateTime) -> bool {
+        !self.is_creation_timestamp_outdated(now) && self.is_oracle_event_id_sane_to_take(now)
+    }
+
+    /// Check if the the maker's offer creation timestamp is outdated
+    ///
+    /// If the creation timestamp is older than 10 minutes the offer is considered outdated
+    fn is_creation_timestamp_outdated(&self, now: OffsetDateTime) -> bool {
+        self.creation_timestamp_maker.seconds() + (Self::OUTDATED_AFTER_MINS * 60)
+            < now.unix_timestamp()
+    }
+
+    /// Check the oracle event ID for sanity
+    ///
+    /// An id within [25h, 23h] from now is considered sane.
+    fn is_oracle_event_id_sane_to_take(&self, now: OffsetDateTime) -> bool {
```

Comment:
🔧 Maybe a bit clearer.

Talking about event ID is always a bit confusing because it's not immediately obvious that the timestamp is embedded in there.

I'm not too fond of the word "sane" because it's generic, but it's hard to express concisely "far enough into the future, but not too far". I guess you could split this condition, but it might not be worth it.

```suggestion
    fn is_oracle_settlement_time_sane(&self, now: OffsetDateTime) -> bool {
```